packages/m2m-gateway/src/services/eserviceTemplateService.ts

```diff
@@ -90,5 +110,303 @@ export function eserviceTemplateServiceBuilder(
+
+      const response =
+        await clients.eserviceTemplateProcessClient.createEServiceTemplate(
+          seed,
+          {
+            headers,
+          }
+        );
+      const polledResource = await pollEServiceTemplate(response, headers);
+      return toM2MGatewayEServiceTemplate(polledResource.data);
+    },
+    async updatePublishedEServiceTemplateDescription(
+      templateId: EServiceTemplateId,
+      seed: m2mGatewayApi.EServiceTemplateDescriptionUpdateSeed,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceTemplate> {
+      logger.info(
+        `Updating description for published eservice template with id ${templateId}`
+      );
+
+      const response =
+        await clients.eserviceTemplateProcessClient.updateEServiceTemplateDescription(
+          seed,
+          {
+            params: { templateId },
+            headers,
+          }
+        );
+      const polledResource = await pollEServiceTemplate(response, headers);
+      return toM2MGatewayEServiceTemplate(polledResource.data);
+    },
+    async updatePublishedEServiceTemplateIntendedTarget(
+      templateId: EServiceTemplateId,
+      seed: m2mGatewayApi.EServiceTemplateIntendedTargetUpdateSeed,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceTemplate> {
+      logger.info(
+        `Updating intended target for published eservice template with id ${templateId}`
+      );
+
+      const response =
+        await clients.eserviceTemplateProcessClient.updateEServiceTemplateIntendedTarget(
+          seed,
+          {
+            params: { templateId },
+            headers,
+          }
+        );
+      const polledResource = await pollEServiceTemplate(response, headers);
+      return toM2MGatewayEServiceTemplate(polledResource.data);
+    },
+    async updatePublishedEServiceTemplateName(
+      templateId: EServiceTemplateId,
+      seed: m2mGatewayApi.EServiceTemplateNameUpdateSeed,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceTemplate> {
+      logger.info(
+        `Updating name for published eservice template with id ${templateId}`
+      );
+
+      const response =
+        await clients.eserviceTemplateProcessClient.updateEServiceTemplateName(
+          seed,
+          {
+            params: { templateId },
+            headers,
+          }
+        );
+      const polledResource = await pollEServiceTemplate(response, headers);
+      return toM2MGatewayEServiceTemplate(polledResource.data);
+    },
+    async createEServiceTemplateVersion(
+      templateId: EServiceTemplateId,
+      { logger, headers }: WithLogger<M2MGatewayAppContext>
+    ): Promise<m2mGatewayApi.EServiceTemplateVersion> {
+      logger.info(
+        `Creating new version for eservice template with id ${templateId}`
+      );
+
+      const createdVersion =
```

Comment:
See what was done for eservice descriptor creation

```suggestion
     const {
        data: { eserviceTemplate, createdVersionId },
        metadata,
      }
```